docs/pages/pmd/languages/java_metrics_index.md

```diff
@@ -62,6 +62,76 @@ public class Foo { // total 8
 
 * Option `includeJavaLang`: Also include classes from the package `java.lang`
 
+## Cognitive Complexity (COGNITIVE_COMPLEXITY)
+
+*Operation metric.* Can be calculated on any non-abstract operation.
+
+### Description
+
+Cognitive complexity is a measure of how difficult it is for humans to read and understand a method. Code that contains
+a break in the control flow is more complex, whereas the use of language shorthands doesn't increase the level of
+complexity. Nested control flows can make a method more difficult to understand, with each additional nesting of the
+control flow leading to an increase in cognitive complexity.
+
+Information about Cognitive complexity can be found in the original paper here:
+[CognitiveComplexity](https://www.sonarsource.com/docs/CognitiveComplexity.pdf)
+
+By default, this rule reports methods with a complexity of 15 or more. Reported methods should be broken down into less
+complex components.
+
+### Basic Idea
+
+1. Ignore structures that allow multiple statements to be readably shorthanded into one
+2. Increment (add one) for each break in the linear flow of the code
+3. Increment when flow-breaking structures are nested
+
+### Increments
+There is an increment for each of the following:
+* `if`, `else if`, `else`, ternary operator
+* `switch`
+* `for`, `foreach`
+* `while`, `do while`
+* `catch`
+* `goto LABEL`, `break LABEL`, `continue LABEL`, `break NUMBER`, `continue NUMBER`
```

Comment:
Is there really a `break NUMBER` statement? I'm not aware of this...
Same for `continue NUMBER`.

Reply:
I see, that's from the specification of the metric... so for Java, there is no `break NUMBER`, but for other languages there could be.

I think, we should remove these two statements, since this page describes how the metric is implemented for java, and I find it a bit irritating.

```suggestion
* `goto LABEL`, `break LABEL`, `continue LABEL`
```